### PR TITLE
PMM-9309: show support tickets

### PR DESCRIPTION
**Pull request by fabio-silva**
**What this PR does / why we need it**: Add the page to show support tickets from the Platform

**Which issue(s) this PR fixes**: https://jira.percona.com/browse/PMM-9309
Tickets

### CODE REVIEW

File: public/app/percona/tickets/TicketsPage.tsx [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+  }, []);
+
+  const redirect = (url: string) => {
+    window.open(url, '_blank');
+  };
+
+  const getRowProps = (row: Row<Ticket>) => ({
+    key: 'row',
+    onClick: () => redirect(row.original.url),
+    className: css`
```
Comment (tiagomotasantos) [inline public/app/percona/tickets/TicketsPage.tsx:86]: Move to styles file

File: public/app/percona/tickets/TicketsPage.tsx [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+      cursor: pointer;
+      &:hover {
+        background-color: ${palette.gray15};
+      }
+    `,
+  });
+
+  const getCellProps = () => ({
+    key: 'cell',
+    className: css`
```
Comment (tiagomotasantos) [inline public/app/percona/tickets/TicketsPage.tsx:96]: Move to styles file

File: public/app/percona/tickets/TicketsPage.test.tsx [hunk 3]
```diff
@@ -0,0 +1,10 @@
+import React from 'react';
+import { render, screen } from '@testing-library/react';
+import TicketsPage from './TicketsPage';
+
+describe('TicketsPage', () => {
+  it('renders PageWrapper', async () => {
+    await render(<TicketsPage />);
+    expect(screen.queryByText('Tickets')).toBeInTheDocument();
```
Comment (tiagomotasantos) [inline public/app/percona/tickets/TicketsPage.test.tsx:8]: Can data-testid be used here? Text can change.

**Review by tiagomotasantos — CHANGES_REQUESTED** [review]
@filipmikes1 can we add an image of how this looks to the PR description please?

File: public/app/percona/tickets/TicketsPage.tsx [hunk 4]
```diff
@@ -0,0 +1,118 @@
...
+        return;
+      }
+      logger.error(e);
+    }
+    setPending(false);
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, []);
+
+  useEffect(() => {
+    getData(true).then(() => triggerTimeout(getData, DATA_INTERVAL));
```
[the remark this answers is not in the document]
Comment (fabio-silva) [reply public/app/percona/tickets/TicketsPage.tsx]: I tested this locally and I agree with Tiago. When not authorised, we keep getting that warning every 5s, it gets very annoying very fast. Let's remove the timeout for now.

[the remark this answers is not in the document]
Comment (filipmikes1) [reply public/app/percona/tickets/TicketsPage.tsx]: I saw it in other parts of application so I thought that it is a good idea to use it. I can remove it no problem.

**Review by tiagomotasantos — APPROVED** [review]
Good job 🚀

File: public/app/percona/tickets/Tickets.styles.ts [hunk 5]
```diff
@@ -1,8 +1,18 @@
 import { GrafanaTheme } from '@grafana/data';
 import { css } from '@emotion/css';
+import { palette } from '@grafana/data/src/themes/palette';
```
Comment (tiagomotasantos) [inline public/app/percona/tickets/Tickets.styles.ts:3]: `palette` is part of `GrafanaTheme` passed as parameter to `getStyles`. No need to import

File: public/app/percona/tickets/TicketsPage.tsx [hunk 6]
```diff
@@ -90,23 +86,16 @@ export const TicketsPage: FC = () => {
...
     key: 'cell',
-    className: css`
-      background-color: transparent !important;
-    `,
+    className: styles.cellProps,
   });
 
   return (
-    <PageWrapper pageModel={PAGE_MODEL}>
+    <PageWrapper pageModel={PAGE_MODEL} dataTestId={'page-wrapper-tickets'}>
```
Comment (tiagomotasantos) [inline public/app/percona/tickets/TicketsPage.tsx:98]: `dataTestId="page-wrapper-tickets"`

**Review by tiagomotasantos — CHANGES_REQUESTED** [review]

File: public/app/percona/tickets/Tickets.constants.ts [hunk 7]
```diff
@@ -0,0 +1,9 @@
+import { PageModel } from 'app/core/components/Breadcrumb';
+
+export const LIST_TICKETS_CANCEL_TOKEN = 'listTickets';
+export const DATA_INTERVAL = 5000;
```
Comment (fabio-silva) [inline public/app/percona/tickets/Tickets.constants.ts:4]: @filipmikes1 please remove this `DATA_INTERVAL`, as we no longer use it

**Review by tiagomotasantos — APPROVED** [review]

File: public/app/percona/tickets/Tickets.messages.ts [hunk 8]
```diff
@@ -0,0 +1,15 @@
...
+  table: {
+    columns: {
+      number: 'Number',
+      description: 'Short Description',
+      priority: 'Priority',
+      state: 'State',
+      createDate: 'Create date',
+      department: 'Department',
+      type: 'Task type',
+      url: 'Url',
```
Comment (fabio-silva) [inline public/app/percona/tickets/Tickets.messages.ts:11]:
```suggestion
      url: 'URL',
```

File: public/app/percona/tickets/Tickets.messages.ts [hunk 9]
```diff
@@ -0,0 +1,15 @@
+export const Messages = {
+  table: {
+    columns: {
+      number: 'Number',
+      description: 'Short Description',
+      priority: 'Priority',
+      state: 'State',
+      createDate: 'Create date',
```
Comment (fabio-silva) [inline public/app/percona/tickets/Tickets.messages.ts:8]:
```suggestion
      createDate: 'Creation Date',
```

File: public/app/percona/tickets/Tickets.messages.ts [hunk 10]
```diff
@@ -0,0 +1,15 @@
...
+      department: 'Department',
+      type: 'Task type',
```
Comment (fabio-silva) [inline public/app/percona/tickets/Tickets.messages.ts:10]:
```suggestion
      type: 'Task Type',
```

File: public/app/percona/tickets/TicketsPage.tsx [hunk 11]
```diff
@@ -0,0 +1,116 @@
...
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [connected]);
+
+  const redirect = (url: string) => {
+    window.open(url, '_blank');
+  };
+
+  const getRowProps = (row: Row<Ticket>) => ({
+    key: 'row',
+    onClick: () => redirect(row.original.url),
```
Comment (fabio-silva) [inline public/app/percona/tickets/TicketsPage.tsx:88]: I believe we can do `window.open(url, '_blank')` right away, no need for a one-line function here.

File: public/app/percona/tickets/TicketsPage.tsx [hunk 12]
```diff
@@ -0,0 +1,116 @@
...
+    className: styles.rowProps,
+  });
+
+  const getCellProps = () => ({
+    key: 'cell',
```
Comment (fabio-silva) [inline public/app/percona/tickets/TicketsPage.tsx:93]: Same as in row. Merge main to your branch.
Then: 
```
const getCellProps = (cell: Cell<Ticket>) => ({
    key: cell.row.original.url,
    className: styles.cellProps,
})
```

**Review by fabio-silva — COMMENTED** [review]
Good job @filipmikes1  🙌 
Just some small changes needed.

File: public/app/percona/tickets/TicketsPage.tsx (hunk 12 above)
Comment (fabio-silva) [reply public/app/percona/tickets/TicketsPage.tsx:93]: P.S. actually, it should be `cell.value`, the URL would be the same for all the cells in a given row

File: public/app/percona/shared/components/Elements/PermissionLoader/PermissionLoader.tsx [hunk 13]
```diff
@@ -0,0 +1,33 @@
...
+
+  return (
+    <div className={styles.emptyBlock}>
+      <EmptyBlock dataTestId="empty-block">
+        {isLoading ? (
+          <Spinner />
+        ) : isAuthorized ? (
+          renderError()
+        ) : (
+          <div data-testid="unauthorized">{Messages.unauthorized}</div>
```
Comment (solovevayaroslavna) [inline public/app/percona/shared/components/Elements/PermissionLoader/PermissionLoader.tsx:28]: I dind't see any changes for tests, and for my opinion it's not really good when we move some of the functions to a separate file, but we don't transfer the tests related to these functions. For example "'should show insufficient access permissions message" test in FeatureLoader.test.tsx

File: public/app/percona/shared/services/user/User.service.ts [hunk 14]
```diff
@@ -0,0 +1,17 @@
+import { CancelToken } from 'axios';
+import { api } from 'app/percona/shared/helpers/api';
+import { UserConnectedResponse } from './User.types';
+
+const BASE_URL = '/v1/Platform';
+
+export const UserService = {
```
Comment (solovevayaroslavna) [inline public/app/percona/shared/services/user/User.service.ts:7]: I also see that the guys usually cover the work of hooks in .service.ts files with unit tests, so it seems to me that this should also be covered by tests, maybe I'm wrong...

**Review by solovevayaroslavna — CHANGES_REQUESTED** [review]

**Review by tiagomotasantos — APPROVED** [review]

File: public/app/percona/shared/services/user/User.service.ts (hunk 14 above)
Comment (fabio-silva) [reply public/app/percona/shared/services/user/User.service.ts:7]: @solovevayaroslavna This service only does an API call, we don't have a hook here or any business logic, if I got this correctly.

Comment (filipmikes1) [reply public/app/percona/shared/services/user/User.service.ts:7]: I agree there is no need for unit test here. Resolving.

File: public/app/percona/shared/components/Elements/PermissionLoader/PermissionLoader.tsx (hunk 13 above)
Comment (filipmikes1) [reply public/app/percona/shared/components/Elements/PermissionLoader/PermissionLoader.tsx:28]: Tests added

**Review by solovevayaroslavna — APPROVED** [review]

**Review by tiagomotasantos — APPROVED** [review]

**Review by filipmikes1 — APPROVED** [review]

**Pull request merged**